src/gfx_layout.cpp

```diff
@@ -123,65 +126,51 @@ static inline void GetLayouter(Layouter::LineCacheItem &line, const char *&str,
  * @param colour   The colour of the font.
  * @param fontsize The size of font to use.
  */
-Layouter::Layouter(const char *str, int maxw, TextColour colour, FontSize fontsize) : string(str)
+Layouter::Layouter(std::string_view str, int maxw, TextColour colour, FontSize fontsize) : string(str)
 {
 	FontState state(colour, fontsize);
-	WChar c = 0;
 
-	do {
-		/* Scan string for end of line */
-		const char *lineend = str;
-		for (;;) {
-			size_t len = Utf8Decode(&c, lineend);
-			if (c == '\0' || c == '\n') break;
-			lineend += len;
-		}
+	while (true) {
+		auto line_length = str.find_first_of('\n');
+		auto str_line = str.substr(0, line_length);
 
-		LineCacheItem& line = GetCachedParagraphLayout(str, lineend - str, state);
+		LineCacheItem& line = GetCachedParagraphLayout(str_line, state);
```

Comment:
```suggestion
		LineCacheItem &line = GetCachedParagraphLayout(str_line, state);
```
It's already wrong... but if you're changing it (and I found more tiny things...)